### PR TITLE
dev-python/jedi: Add v0.14.1

### PR DESCRIPTION
**Pull request by AndrewAmmerlaan**
\>=dev-python/spyder-4.0.0 (https://github.com/gentoo/gentoo/pull/13988) requires =dev-python/jedi-0.14.1

Package-Manager: Portage-2.3.85, Repoman-2.3.20
Signed-off-by: Andrew Ammerlaan <andrewammerlaan@riseup.net>

### CODE REVIEW

**Comment by AndrewAmmerlaan** [comment]
@juippis what is going here?

>There are too many broken packages to determine whether the breakages were added by the pull request. If in doubt, please rebase.

I did rebase just now, I'm only 7 commits behind (at the time of writing)

**Comment by juippis** [comment]
> 
> > There are too many broken packages to determine whether the breakages were added by the pull request. If in doubt, please rebase.
> 

This is usually caused by a broken tree state which is often fastly fixed - however now there are simply too many CI issues with upstream ::gentoo repo, that needs to be fixed first.

**Comment by juippis** [comment]
^ I do see many packages in there that have been fixed already though, so maybe you should fetch the latest state and give it one more try...

**Comment by AndrewAmmerlaan** [comment]
> ^ I do see many packages in there that have been fixed already though, so maybe you should fetch the latest state and give it one more try...
> 
> EDIT: .... but dont do it now, as the tree is actually broken.
> https://archives.gentoo.org/gentoo-automated-testing/threads/2020-01/
> 
> What fun times!
> 
> EDITEDIT: Seems fixed again.

Alright, I just rebased and pushed my PRs. Hopefully the bot won't complain now

**Comment by juippis** [comment]
I can't get the tests to run with this, are you sure this version is needed? Won't spyder work with 0.15?

**Comment by AndrewAmmerlaan** [comment]
> I can't get the tests to run with this, are you sure this version is needed? Won't spyder work with 0.15?

Version 0.15 works but you get a very annoying popup whenever you start spyder stating that you have the wrong version. Apparently there is an issue with completion for numpy and pandas in 0.15 and up.

What error are you getting? I'm pretty sure tests were working for me yesterday, I'll check again.

**Comment by AndrewAmmerlaan** [comment]
> I can't get the tests to run with this, are you sure this version is needed? Won't spyder work with 0.15?

Tests work for me:
```
======================== 2378 passed, 34 skipped, 5 xfailed in 46.26 seconds ===========================
```

Maybe I'm missing a dependency in the ebuild?

**Comment by juippis** [comment]
```
        # Ensure that pth and egg-link paths were added.
>       assert venv_paths[-len(ETALON):] == ETALON
E       AssertionError: assert ['/tmp/portag...py:from_func'] == ['/tmp/portage...py:from_func']
E         At index 0 diff: '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages' != '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages'
E         Full diff:
E         - ['/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages',
E         ?                                                                               --
E         + ['/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages',
E         -  '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages/dir-from-foo-pth',
E         ?                                                                               --
E         +  '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages/dir-from-foo-pth',
E         '/foo/smth.py:module',
E         '/foo/smth.py:from_func',
E         '/foo/smth.py:from_func']

test/test_evaluate/test_sys_path.py:61: AssertionError
=========== 1 failed, 2366 passed, 45 skipped, 5 xfailed in 49.87 seconds ============
 * ERROR: dev-python/jedi-0.14.1::testworld failed (test phase):
 *   Tests fail with python3.6
```
I'm currently on my small laptop so I don't want to investigate this much, but yeah, I guess 0.14.1 is needed then.

**Comment by AndrewAmmerlaan** [comment]
> ```
>         # Ensure that pth and egg-link paths were added.
> >       assert venv_paths[-len(ETALON):] == ETALON
> E       AssertionError: assert ['/tmp/portag...py:from_func'] == ['/tmp/portage...py:from_func']
> E         At index 0 diff: '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages' != '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages'
> E         Full diff:
> E         - ['/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages',
> E         ?                                                                               --
> E         + ['/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages',
> E         -  '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages/dir-from-foo-pth',
> E         ?                                                                               --
> E         +  '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages/dir-from-foo-pth',
> E         '/foo/smth.py:module',
> E         '/foo/smth.py:from_func',
> E         '/foo/smth.py:from_func']
> 
> test/test_evaluate/test_sys_path.py:61: AssertionError
> =========== 1 failed, 2366 passed, 45 skipped, 5 xfailed in 49.87 seconds ============
>  * ERROR: dev-python/jedi-0.14.1::testworld failed (test phase):
>  *   Tests fail with python3.6
> ```
> 
> I'm currently on my small laptop so I don't want to investigate this much, but yeah, I guess 0.14.1 is needed then.

Aaah, Thank you. 

I was compiling the ebuild in py3_7, that test seems to work in that version. However, I found this in the 0.12.1 ebuild:

```
	# our very useful patching changes libdir for no good reason
	sed -i -e "/site_pkg_path/s:'lib':& if virtualenv.version_info >= (3,7) else '$(get_libdir)':" \
		test/test_evaluate/test_sys_path.py || die
``` 
I think that this is supposed to prevent this very problem, I'll try implementing a similar fix in the 0.14.1 ebuild.

**Comment by AndrewAmmerlaan** [comment]
> ```
>         # Ensure that pth and egg-link paths were added.
> >       assert venv_paths[-len(ETALON):] == ETALON
> E       AssertionError: assert ['/tmp/portag...py:from_func'] == ['/tmp/portage...py:from_func']
> E         At index 0 diff: '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages' != '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages'
> E         Full diff:
> E         - ['/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages',
> E         ?                                                                               --
> E         + ['/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages',
> E         -  '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib64/python3.6/site-packages/dir-from-foo-pth',
> E         ?                                                                               --
> E         +  '/tmp/portage/dev-python/jedi-0.14.1/temp/pytest-of-portage/pytest-0/venv/lib/python3.6/site-packages/dir-from-foo-pth',
> E         '/foo/smth.py:module',
> E         '/foo/smth.py:from_func',
> E         '/foo/smth.py:from_func']
> 
> test/test_evaluate/test_sys_path.py:61: AssertionError
> =========== 1 failed, 2366 passed, 45 skipped, 5 xfailed in 49.87 seconds ============
>  * ERROR: dev-python/jedi-0.14.1::testworld failed (test phase):
>  *   Tests fail with python3.6
> ```
> 
> I'm currently on my small laptop so I don't want to investigate this much, but yeah, I guess 0.14.1 is needed then.

How about now? 

This specific test changed between versions 0.12 and 0.14 so I cannot implement the same fix. I would need to undo some of the changes that upstream made between these versions. i suspect these changes were made for a good reason, so I'd rather not do that. Instead I skip the test only for python3_6.

File: dev-python/jedi/jedi-0.14.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	# this super-secret feature of py3.4 apparently doesn't work for us
+	sed -i -e 's:test_init_extension_module:_&:' \
+		test/test_evaluate/test_extension.py || die
+
+	distutils-r1_python_prepare_all
+}
+
+python_test() {
+	if [ "${EPYTHON}" = "python3.6" ]; then
```
Comment (juippis) [inline dev-python/jedi/jedi-0.14.1.ebuild]: `if [[ ${EPYTHON} = python3.6 ]]; then` should be "better bash" these days.

File: dev-python/jedi/jedi-0.14.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	sed -i "s:'docopt',:: ; s:'colorama',::" setup.py || die
+	sed -i "s: --doctest-modules::" pytest.ini || die
+
+	# no clue why it fails but we don't really care about .pyc files
+	# without sources anyway
+	rm test/test_evaluate/test_pyc.py || die
+
+	# this super-secret feature of py3.4 apparently doesn't work for us
+	sed -i -e 's:test_init_extension_module:_&:' \
+		test/test_evaluate/test_extension.py || die
```
Comment (juippis) [inline dev-python/jedi/jedi-0.14.1.ebuild]: Seems useless by the way.

Comment (AndrewAmmerlaan) [reply dev-python/jedi/jedi-0.14.1.ebuild]: Indeed, it seems to work exactly the same without that line, so I have removed it

**Pull request closed without merging**